.github/workflows/documentation-workflow.yml

```diff
@@ -0,0 +1,24 @@
+name: Documentation
+
+on:
+  push:
+    branches:
+      - master
+    tags: 'v*'
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: julia-actions/setup-julia@latest
+        with:
+          version: '1.4'
+      - name: Install dependencies
+        run: julia --project=docs/ -e 'using Pkg; Pkg.develop(PackageSpec(path=pwd())); Pkg.instantiate(); Pkg.build("IJulia")'
+      - name: Build and deploy
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # For authentication with GitHub Actions token
```

Comment:
Seems like we will run into https://github.com/JuliaDocs/Documenter.jl/issues/1177?

Reply:
Seems like the problem is with initially creating the `gh-pages` branch, the following comment says you can get around it by making an initial (empty) commit to it:
https://github.com/JuliaDocs/Documenter.jl/issues/1177#issuecomment-640998242

Reply:
Okay, I created an empty `gh-pages` branch.